src/DotLiquid/Hash.cs

```diff
@@ -240,22 +241,22 @@ public void Clear()
 
         IDictionaryEnumerator IDictionary.GetEnumerator()
         {
-            return ((IDictionary) _nestedDictionary).GetEnumerator();
+            return ((IDictionary)_nestedDictionary).GetEnumerator();
         }
 
         public bool Contains(KeyValuePair<string, object> item)
         {
-            return ((IDictionary<string, object>) _nestedDictionary).Contains(item);
+            return _defaultValue != null || _lambda != null || ((IDictionary<string, object>)_nestedDictionary).Contains(item);

```

Comment:
This change doesnt seem correct. It doesn't contain that key/value pair only that key especially is defaultValue is not equal to the value of kvp

Reply:
Removed _defaultValue!=null check for all Contain(...) methods.

Reply:
Change reverted, new tests retained to document pre-existing behaviour.